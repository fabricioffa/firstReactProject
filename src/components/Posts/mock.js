export const postsPropsMock = [
  {
    id: 1,
    title: 'A title 1',
    cover: 'img/img1.img',
    body: 'A body 1',
  },
  {
    id: 2,
    title: 'A title 2',
    cover: 'img/img2.img',
    body: 'A body 2',
  },
  {
    id: 3,
    title: 'A title 3',
    cover: 'img/img3.img',
    body: 'A body 3',
  },
];
