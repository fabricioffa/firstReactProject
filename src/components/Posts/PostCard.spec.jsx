import { render, screen } from '@testing-library/react';
import { Posts } from '.';
import { postsPropsMock as props } from './mock';

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts posts={props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /title 3/i })).toHaveAttribute('src', 'img/img3.img');
  });

  it('should not render posts', () => {
    render(<Posts />);

    expect(screen.queryAllByRole('heading', { name: /title/i })).toHaveLength(0);

    // Or
    // expect(screen.queryByRole("heading", { name: /title/i })).not.toBeInTheDocument(q);
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts posts={props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
