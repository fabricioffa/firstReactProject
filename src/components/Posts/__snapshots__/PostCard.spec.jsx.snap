// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="A title 1"
      src="img/img1.img"
    />
    <div
      class="post-content"
    >
      <h2>
        A title 1
      </h2>
      <p>
        A body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="A title 2"
      src="img/img2.img"
    />
    <div
      class="post-content"
    >
      <h2>
        A title 2
      </h2>
      <p>
        A body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="A title 3"
      src="img/img3.img"
    />
    <div
      class="post-content"
    >
      <h2>
        A title 3
      </h2>
      <p>
        A body 3
      </p>
    </div>
  </div>
</div>
`;
